### PR TITLE
Add linesep kwarg to Paginator

### PR DESCRIPTION
**Pull request by PikalaxALT**
## Summary

Extends the commands.Paginator class to allow custom line separators, for example if you want to
- add individual words rather than whole lines (useful if your input material is all on one line)
- make an Emoji paginator with no spaces between Emoji objects
- clap 👏 at 👏 your 👏 friends

## Checklist

- [x] If code changes were made then they have been tested.
    - [x] I have updated the documentation to reflect the changes.
- [ ] This PR fixes an issue.
- [x] This PR adds something new (e.g. new method or parameters).
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed)
- [ ] This PR is **not** a code change (e.g. documentation, README, ...)

### CODE REVIEW

File: discord/ext/commands/help.py [hunk 1]
```diff
@@ -162,8 +169,8 @@ def pages(self):
         return self._pages
 
     def __repr__(self):
-        fmt = '<Paginator prefix: {0.prefix} suffix: {0.suffix} max_size: {0.max_size} count: {0._count}>'
-        return fmt.format(self)
+        fmt = '<Paginator prefix: {0.prefix} suffix: {0.suffix} linesep: {1} max_size: {0.max_size} count: {0._count}>'
+        return fmt.format(self, repr(self.linesep))
```
Comment (Rapptz) [inline discord/ext/commands/help.py:173]:
```suggestion
        fmt = '<Paginator prefix: {0.prefix} suffix: {0.suffix} linesep: {0.linesep!r} max_size: {0.max_size} count: {0._count}>'
        return fmt.format(self)
```

File: discord/ext/commands/help.py [hunk 2]
```diff
@@ -79,18 +79,21 @@ class Paginator:
         The suffix appended at the end of every page. e.g. three backticks.
     max_size: :class:`int`
         The maximum amount of codepoints allowed in a page.
+    linesep: :class:`str`
+        The character string inserted between lines. e.g. a newline character.
```
Comment (Rapptz) [inline discord/ext/commands/help.py:83]: Needs a versionadded string here.

**Review by Rapptz — CHANGES_REQUESTED** [review]
Truthfully I'm a bit unsure whether there's much use case that makes this PR worth merging. The interface in general assumes that things are line separated both in naming and in the original code and I don't know how common this operation is.

File: discord/ext/commands/help.py (hunk 1 above)
Comment (PikalaxALT) [reply discord/ext/commands/help.py:173]: Might as well repr the prefix and suffix too while I'm at it.

**Review by Rapptz — APPROVED** [review]
I guess this is fine. Though admittedly I'm still not sure I get the use cases.

**Pull request merged**